### PR TITLE
Sign out from current space

### PR DESCRIPTION
**Pull request by cp-pratik-k**
### Purpose
 - Fix leave type not shown 
 - User sould be able to sign out with space and navigate join space.

### Summary of Changes
 - Fix some ui issues 
 - Add sign out with space button
 - Add sign out in join space for sign out with current account
 - Update test according new change

### Test steps
 - Run flutter test

### Conformity
- [x] Followed [git guidelines](https://github.com/canopas/flutter-developer-roadmap/blob/main/GitGuideline.md) for creating commit messages and Pull Request guidelines.
- [x] Self-approved the PR - reviewed the PR as a reviewer and gave it self-approval if everything is ok. If not, made the required changes.
- [x] Ensured that the PR satisfies all specified requirements in the ticket, including bug fixes and new features.
- [x] Provided test steps, including steps to reproduce the issue or test the new functionality, ensuring other team members can verify the changes.
- [x] Added/Updated proper code comments to make it easy-to-understand for other developers.
- [x] Reused code (if the same code was written twice, made it common and reused it at both places).
- [x] Removed unused or commented code if not required.
- [x] Ensured proper Dart naming conventions were used for variables, classes, and methods.
- [x] Localized user-facing strings.
- [x] Included screenshots/videos of behavior changes: Provided visual evidence of any changes to UI or behavior for easier review and understanding in the PR description.
- [x] Implemented proper error handling: Ensured that the code anticipated and handled potential errors and edge cases gracefully.
- [x] Avoided introducing technical debt: If the PR introduces technical debt, created and linked appropriate tickets for future resolution.
- [x] Included relevant unit tests: Wrote unit tests that focused on testing behavior and functionality, rather than merely covering lines of code.
- [x] Ensured code was performant and scalable: Verified that the changes did not introduce performance issues or bottlenecks and could scale as needed.
- [x] Ensured comments were up-to-date and relevant to the code to describe complex logic and to add understanding for other developers.
- [x] Marked the PR as ready before submitting it for review.

### CODE REVIEW

File: lib/ui/shared/appbar_drawer/drawer/app_drawer.dart [hunk 1]
```diff
@@ -112,25 +116,20 @@ class DrawerOptionList extends StatelessWidget {
...
-                    title: locale.sign_out_tag,
-                    description: locale.sign_out_alert,
-                  );
-                }),
+              icon: Icons.logout_rounded,
+              title: locale.sign_out_from_text(userStateNotifier.currentSpace?.name ?? "current space"),
+              onTap: () => context
+                  .read<DrawerBloc>()
+                  .add(SignOutWithCurrentSpaceEvent()),
+            ),
```
Comment (cp-sneha-s) [inline lib/ui/shared/appbar_drawer/drawer/app_drawer.dart:132]: l`ocale.sign_out_from_text(userStateNotifier.currentSpace?.name ?? "current space"`
No need to add Null check here as name can't be empty for space.

File: lib/ui/shared/appbar_drawer/drawer/bloc/app_drawer_event.dart [hunk 2]
```diff
@@ -10,5 +10,4 @@ class ChangeSpaceEvent extends DrawerEvents {
   ChangeSpaceEvent(this.space);
 }
 
-class SignOutEvent extends DrawerEvents {
-}
+class SignOutWithCurrentSpaceEvent extends DrawerEvents {}
```
Comment (cp-sneha-s) [inline lib/ui/shared/appbar_drawer/drawer/bloc/app_drawer_event.dart:13]: Rename `SignOutWithCurrentSpaceEvent` -> `SignOutFromSpace`

File: lib/ui/space/join_space/bloc/join_space_bloc.dart [hunk 3]
```diff
@@ -117,4 +120,20 @@ class JoinSpaceBloc extends Bloc<JoinSpaceEvents, JoinSpaceState> {
...
+      if (isLogOut) {
+        await _userManager.removeAll();
+        emit(state.copyWith(signOutStatus: Status.success));
+      } else {
+        throw Exception(signOutError);
+      }
+    } on Exception {
+      emit(state.copyWith(error: signOutError, signOutStatus: Status.error));
+    }
+  }
```
Comment (cp-sneha-s) [inline lib/ui/space/join_space/bloc/join_space_bloc.dart:138]: What is the signoutError? Is it managed well for UI?

File: lib/ui/space/join_space/bloc/join_space_event.dart [hunk 4]
```diff
@@ -24,4 +20,6 @@ class SelectInvitationEvent extends JoinSpaceEvents {
   final Invocation invitation;
 
   SelectInvitationEvent({required this.invitation});
-}
```
Comment (cp-sneha-s) [inline lib/ui/space/join_space/bloc/join_space_event.dart:27]: Rename `Invocation` with `Invitation`.

File: lib/ui/space/join_space/bloc/join_space_bloc.dart (hunk 3 above)
Comment (cp-pratik-k) [reply lib/ui/space/join_space/bloc/join_space_bloc.dart:138]: yet it is already implemented

"Your attempt to sign out failed,Please try again later"

File: lib/ui/shared/appbar_drawer/drawer/app_drawer.dart [hunk 5]
```diff
@@ -125,10 +125,10 @@ class DrawerOptionList extends StatelessWidget {
                 previous.signOutStatus != current.signOutStatus,
             builder: (context, state) => DrawerOption(
               icon: Icons.logout_rounded,
-              title: locale.sign_out_from_text(userStateNotifier.currentSpace?.name ?? "current space"),
+              title: locale.sign_out_from_text(userStateNotifier.currentSpace!.name),
               onTap: () => context
                   .read<DrawerBloc>()
-                  .add(SignOutWithCurrentSpaceEvent()),
+                  .add(SignOutFromSpace()),
             ),
```
Comment (cp-sneha-s) [inline lib/ui/shared/appbar_drawer/drawer/app_drawer.dart]: Sorry, Add **Event** Extension like `SignOutFromSpaceEvent`

**Review by cp-sneha-s — COMMENTED** [review]
Let's change `Signout from pratik.k@cnopas.com` to only `Signout`

**Comment by cp-pratik-k** [comment]
> Let's change `Signout from pratik.k@cnopas.com` to only `Signout` Also, Add Elevated button for this
this look good instead off elevented button and i think we no need to check ui now as we create new UI.

**Comment by cp-pratik-k** [comment]
@cp-sneha-s Let's review in priority.

**Comment by cp-sneha-s** [comment]
> Collaborator

Let's add only `Sign Out` instead of `signout from pratik.k@canopas.com`

**Review by cp-sneha-s — APPROVED** [review]
